test/junit/scala/math/BigDecimalTest.scala

```diff
@@ -228,4 +228,35 @@ class BigDecimalTest {
   def test_SI8970() {
     assert((0.1).## == BigDecimal(0.1).##)
   }
+
+  // Motivated by the problem of MathContext lost
+  @Test
+  def testMathContext() {
+    def testPrecision() {
+      val e = 1000
```

Comment:
This is an unfortunate choice of variable name--it makes one think of Euler's number.  Can you change it (e.g. to `p` or `x`)?